tchannel/tornado/peer.py

```diff
@@ -618,7 +617,8 @@ def send_with_retry(self, request, peer, retry_limit, connection):
             try:
                 response = yield self._send(connection, request)
                 raise gen.Return(response)
-            except (ProtocolError, TimeoutError) as error:
+            # Why are we retying on all errors????
+            except TChannelError as error:
```

Comment:
I mean internally. for example since we not have different error type, ideally doesn't need the code to tell which one to retry.